galera/src/replicator_smm_params.cpp

```diff
@@ -126,9 +125,30 @@ galera::ReplicatorSMM::ParseOptions::ParseOptions(Replicator&       repl,
                                                   const char* const opts)
 {
     conf.parse(opts);
-    // Set initial wsrep params here to enable debug logging etc
-    // for the rest of the initialization
-    wsrep_set_params(repl, opts);
+
+    if (conf.get<bool>(Replicator::Param::debug_log))
+    {
+        gu_conf_debug_on();
+    }
+    else
+    {
+        gu_conf_debug_off();
+    }
+#ifdef GU_DBUG_ON
+    if (conf.is_set(galera::Replicator::Param::dbug))
+    {
+        GU_DBUG_PUSH(conf.get(galera::Replicator::Param::dbug).c_str());
+    }
+    else
+    {
+        GU_DBUG_POP();
+    }
+
+    if (conf.is_set(galera::Replicator::Param::signal))
+    {
+        gu_debug_sync_signal(conf.get(galera::Replicator::Param::signal));
+    }
+#endif /* GU_DBUG_ON */
 }
```

Comment:
This is not exactly a duplicate. Even though it looks very similar, I failed to come up with a common function that would satisfy both cases and at the same time not be utterly confusing as to what it is doing.